src/Features/LanguageServer/Protocol/Handler/InlayHint/InlayHintHandler.cs

```diff
@@ -91,7 +91,7 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(InlayHintParams request)
                 inlayHints.Add(inlayHint);
             }
 
-            return inlayHints.ToArray();
+            return inlayHints.MoveToArray();
```

Comment:
we also support moving out the pain array, for the LSP case (which generally operates with normal arrays, not immutable ones).

Reply:
Wouldn't just new'ing it up directly here be simpler?